sktime/forecasting/model_selection/_split.py

```diff
@@ -29,10 +29,17 @@
 from sktime.utils.validation import (
     ACCEPTED_WINDOW_LENGTH_TYPES,
     NON_FLOAT_WINDOW_LENGTH_TYPES,
+    array_is_datetime64,
```

Comment:
due to repetition - would it make sense to refactor these into a single function, with `datetime64` etc being parameters? Or would that be ugly?

Reply:
well, these type checkers are due to be extended with more and more supported types. for starters I added only `datetime64`, but later it could be also `pd.Timestamp`. so, I would prefer to keep this list hardcoded somewhere and a function that checks an argument belonging to this list.